crates/biome_js_analyze/src/lint/nursery/use_explicit_test_assertions.rs

```diff
@@ -0,0 +1,249 @@
+        }
+    }
+}
+
+fn default_assert_function_names() -> Option<Vec<String>> {
+    Some(vec!["expect".to_string(), "assert".to_string()])
+}
+
+fn default_additional_test_block_function_names() -> Option<Vec<String>> {
+    Some(vec!["test".to_string(), "it".to_string()])
+}
+
+impl Rule for UseExplicitTestAssertions {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseExplicitTestAssertionsOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let options = ctx.options();
+
+        // Is this function named it() or test()?
+        if !is_matching_fn(node, &options.additional_test_block_function_names) {
+            return None;
+        }
+
+        let [Some(second)] = node.arguments().ok()?.get_arguments_by_index([1]) else {
+            return None;
+        };
+        let Some(second_expr) = second.as_any_js_expression() else {
+            return None;
+        };
+        let try_test_body = match second_expr {
+            AnyJsExpression::JsArrowFunctionExpression(function) => {
+                if let Some(function_body) = function.body().ok() {
+                    function_body;
+                }
+                return None;
+            }
+            // TODO: support function
+            _ => None,
+        };
+        if let Some(test_body) = try_test_body {
+            if fn_body_contains_call(test_body, &options.assert_function_names) {
+                Some(());
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.syntax().text_trimmed_range(),
+                markup! {
+                    "Missing assertion in test body."
+                },
+            )
+            .note(markup! {
+                "This prevents false positives where a test always passes but isn’t actually testing anything."
+            }).note(markup! {
+                "Add an expect() (Vitest/Jest) or assert() (node:assert) assertion to this test."
+            }),
+        )
+    }
+}
+
+/// Filter functions by name
+fn is_matching_fn(node: &JsCallExpression, names: &Option<Vec<String>>) -> bool {
+    let Some(callee) = node.callee().ok() else {
+        return false;
+    };
+    if let Some(function_name) = callee.get_callee_member_name() {
+        match names {
+            Some(names) => names.iter().any(|s| s == &function_name.to_string()),
+            None => false,
+        };
+    }
+
+    false
+}
+
+/// Detect if a function body has a call expression in given names list
+fn fn_body_contains_call(node: &JsFunctionBody, names: &Option<Vec<String>>) -> bool {
+    node.statements().iter().any(|statement| match statement {
```

Comment:
Help! I am bad at rust and am not sure why I’m getting this error here:

```
no method named `iter` found for struct `JsStatementList` in the current scope
items from traits can only be used if the trait is in scoperustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20[6]?6#file:///Users/drew/Sites/biomejs/biome/crates/biome_js_analyze/src/lint/nursery/use_explicit_test_assertions.rs)

mod.rs(400, 8): the method is available for `JsStatementList` here
```

![CleanShot 2025-05-12 at 21 36 27@2x](https://github.com/user-attachments/assets/1519f785-5128-4190-a17b-24aea7afecc3)

Reply:
🤔 in the playground I know this node has `.statements()`. And I see the `.statements().iter()` code in other parts of the codebase! I know I’m probably missing something basic but I’m confused.

Reply:
You will need to import the `biome_rowan::AstNodeList` trait to call a method defined in the trait. `<expr>.iter()` here is a syntax sugar of `AstNodeList::iter(<expr>)` and rustc needs to know which trait has the `.iter()` method to call it

Reply:
To back up a bit, I’m trying to plan on deep-crawling a function body, because it may be a part of an inner loop, or it may even be a reference to another function declared elsewhere. So if `.statements()` is the wrong way to recursively scan a callback, would love suggestions 🙏

Reply:
> You will need to import the `biome_rowan::AstNodeList`

Ah thank you! That did the trick 🙂